jscomp/runtime/js_string.ml

```diff
@@ -106,326 +101,363 @@ external get : t -> int -> t = ""
 ]}
 *)
-external includes : t -> bool = "includes"
-[@@mel.send.pipe: t]
-(** ES2015 *)
 
+external includesFrom : t -> sub:t -> pos:int -> bool = "includes"
+[@@mel.send]
 (**
-  [includes searchValue start s] returns [true] if [searchValue] is found anywhere within [s] starting at character number [start] (where 0 is the first character), [false] otherwise.
+  [includes s ~sub:searchValue ~start] returns [true] if [searchValue] is found
+  anywhere within [s] starting at character number [start] (where 0 is the
+  first character), [false] otherwise.
 
 {[
-  includesFrom "gram" 1 "programmer" = true;;
-  includesFrom "gram" 4 "programmer" = false;;
-  includesFrom {js|한|js} 1 {js|대한민국|js} = true;;
+  includesFrom "programmer" ~sub:"gram" ~pos:1 = true;;
+  includesFrom "programmer" ~sub:"gram" ~pos:4 = false;;
+  includesFrom {js|대한민국|js} ~sub:{js|한|js} ~pos:1 = true;;
 ]}
 *)
-external includesFrom : t -> int -> bool = "includes"
-[@@mel.send.pipe: t]
-(** ES2015 *)
 
-external indexOf : t -> int = "indexOf"
-[@@mel.send.pipe: t]
+external indexOf : t -> sub:t -> int = "indexOf"
+[@@mel.send]
 (**
-  [indexOf searchValue s] returns the position at which [searchValue] was first found within [s], or [-1] if [searchValue] is not in [s].
+  [indexOf s ~sub:searchValue ] returns the position at which [searchValue] was
+  first found within [s], or [-1] if [searchValue] is not in [s].
 
 {[
-  indexOf "ok" "bookseller" = 2;;
-  indexOf "sell" "bookseller" = 4;;
-  indexOf "ee" "beekeeper" = 1;;
-  indexOf "xyz" "bookseller" = -1;;
+  indexOf "bookseller" ~sub:"ok" = 2;;
+  indexOf "bookseller" ~sub:"sell" = 4;;
+  indexOf "beekeeper" ~sub:"ee" = 1;;
+  indexOf "bookseller" ~sub:"xyz" = -1;;
 ]}
 *)
 
-external indexOfFrom : t -> int -> int = "indexOf"
-[@@mel.send.pipe: t]
-(**
-  [indexOfFrom searchValue start s] returns the position at which [searchValue] was found within [s] starting at character position [start], or [-1] if [searchValue] is not found in that portion of [s]. The return value is relative to the beginning of the string, no matter where the search started from.
+external indexOfFrom : t -> sub:t -> pos:int -> int = "indexOf"
+[@@mel.send]
+(** [indexOfFrom s ~sub:searchValue ~start] returns the position at which
+    [searchValue] was found within [s] starting at character position [start],
+    or [-1] if [searchValue] is not found in that portion of [s]. The return
+    value is relative to the beginning of the string, no matter where the
+    search started from.
 
 {[
-  indexOfFrom "ok" 1 "bookseller" = 2;;
-  indexOfFrom "sell" 2 "bookseller" = 4;;
-  indexOfFrom "sell" 5 "bookseller" = -1;;
-  indexOf "xyz" "bookseller" = -1;;
+  indexOfFrom "bookseller" ~sub:"ok" ~pos:1 = 2;;
+  indexOfFrom "bookseller" ~sub:"sell" ~pos:2 = 4;;
+  indexOfFrom "bookseller" ~sub:"sell" ~pos:5 = -1;;
 ]}
 *)
 
-external lastIndexOf : t -> int = "lastIndexOf"
-[@@mel.send.pipe: t]
+external lastIndexOf : t -> sub:t -> int = "lastIndexOf"
```

Comment:
s/`sub`/`searchString`?

Reply:
StdLabels also uses `sub`, so I figured it'd be easier for OCaml folks.

Reply:
This is a strange framework to me, considering they are bindings to JS APIs, not a Stdlib.

Reply:
OK. I don't have a strong opinion. Let me know what you think of https://github.com/melange-re/melange/pull/908